### PR TITLE
feat(ds): Add a project option for the target sample rate in manual mode

### PR DESCRIPTION
**Pull request by jan-auer**
Closes https://github.com/getsentry/projects/issues/187

### CODE REVIEW

File: src/sentry/api/endpoints/organization_details.py [hunk 1]
```diff
@@ -389,10 +389,6 @@ def validate_targetSampleRate(self, value):
                 "Organization does not have the custom dynamic sample rate feature enabled."
             )
 
-        if not 0.0 <= value <= 1.0:
-            raise serializers.ValidationError(
-                "The targetSampleRate option must be in the range [0:1]"
-            )
```
Comment (jan-auer) [inline src/sentry/api/endpoints/organization_details.py:395]: This is now handled by the serializer

File: src/sentry/projectoptions/defaults.py [hunk 2]
```diff
@@ -184,3 +184,6 @@
     key="sentry:loader_available_sdk_versions",
     epoch_defaults={1: ["8.x", "7.x", "6.x", "5.x", "4.x"], 11: ["8.x", "7.x"]},
 )
+
+# Dynamic sampling rate in project-level "manual" configuration mode
+register(key="sentry:target_sample_rate", default=1.0)
```
Comment (jan-auer) [inline src/sentry/projectoptions/defaults.py:189]: Technically, we wouldn't need a default as we could default this in the DS logic. However, to be more consistent and provide proper values to the UI, I'm adding this default here.

File: src/sentry/api/endpoints/project_details.py [hunk 3]
```diff
@@ -423,6 +425,28 @@ def validate_sensitiveFields(self, value):
...
+        from sentry import features
+
+        organization = self.context["project"].organization
+        actor = self.context["request"].user
+        if not features.has("organizations:dynamic-sampling-custom", organization, actor=actor):
+            raise serializers.ValidationError(
+                "Organization does not have the custom dynamic sample rate feature enabled."
+            )
+
+        if organization.get_option("sentry:sampling_mode", SAMPLING_MODE_DEFAULT) != "project":
```
Comment (vgrozdanic) [inline src/sentry/api/endpoints/project_details.py:438]: nit: Do we want to unite this with [choices](https://github.com/getsentry/sentry/blob/bc8d10ea18505efff605bfae3833ff82b8103809/src/sentry/api/endpoints/organization_details.py#L288) for `samplingMode` from `src/sentry/api/endpoints/organization_details.py` and have a single source of truth which modes are possible?

File: src/sentry/api/endpoints/project_details.py [hunk 4]
```diff
@@ -423,6 +425,28 @@ def validate_sensitiveFields(self, value):
...
+            raise serializers.ValidationError(
+                "Must enable Manual Mode to configure project sample rates."
+            )
+
+        if not 0.0 <= value <= 1.0:
+            raise serializers.ValidationError(
+                "The targetSampleRate option must be between 0 and 1."
+            )
```
Comment (vgrozdanic) [inline src/sentry/api/endpoints/project_details.py:446]:
```suggestion
        if not 0.0 <= value <= 1.0:
            raise serializers.ValidationError(
                "The targetSampleRate option must be between 0 and 1."
            )
```

Isn't this also checked by a serializer?

**Review by vgrozdanic — APPROVED** [review]

**Pull request merged**